src/csvcubeddevtools/behaviour/csv2rdf.py

```diff
@@ -67,39 +82,82 @@ def _run_csv2rdf(metadata_file_path: Path) -> Tuple[int, str, Optional[str]]:
             else:
                 ttl_out = ""
 
-            return status_code, log, ttl_out
+            return Csv2RdfResult(
+                status_code=status_code,
+                log=log,
+                ttl=ttl_out,
+                rdf_input_directory=metadata_file_path.resolve().parent,
+            )
+
+    if tmp_dir is None:
+        with TemporaryDirectory() as tmp_dir:
+            tmp_dir = Path(tmp_dir)
+            return _run_csv2rdf_internal(tmp_dir)
+    else:
+        return _run_csv2rdf_internal(tmp_dir)
 
 
 @step('csv2rdf on "{file}" should succeed')
 def step_impl(context, file: str):
     """csv2rdf on "{file}" should succeed"""
     temp_dir = get_context_temp_dir_path(context)
-    exit_code, logs, ttl_out = _run_csv2rdf(temp_dir / file)
-    assert exit_code == 0
+    csv2rdf_result = _run_csv2rdf(temp_dir / file)
+    assert csv2rdf_result.status_code == 0, csv2rdf_result.log
```

Comment:
Can we not assert anything specific inside the log? For example, if the csv2rdf is success, do we store a status code in the log?

Reply:
This isn't an individual test, these are step definitions being run by other tests. There is no single thing I can assert here beyond that it succeeded.